M365/MDO/MDOThreatPolicyChecker.ps1

```diff
@@ -0,0 +1,894 @@
+#Requires -Modules ExchangeOnlineManagement -Version 3.0.0
+
+<#
+.SYNOPSIS
+Evaluates user coverage and potential redundancies in Microsoft Defender for Office 365 and Exchange Online Protection threat policies, including anti-malware, anti-phishing, and anti-spam policies, as well as Safe Attachments and Safe Links policies if licensed.
+
+.DESCRIPTION
+This script checks which Microsoft Defender for Office 365 and Exchange Online Protection threat policies cover a particular user, including anti-malware, anti-phishing, inbound and outbound anti-spam, as well as Safe Attachments and Safe Links policies in case these are licensed for your tenant. In addition, the script can check for threat policies that have inclusion and/or exclusion settings that may be redundant or confusing and lead to missed coverage of users or coverage by an unexpected threat policy.
+
+.PARAMETER CsvFilePath
+  Allows you to specify a CSV file with a list of email addresses to check.
+.PARAMETER EmailAddresses
+  Allows you to specify email address or multiple addresses separated by commas.
+.PARAMETER IncludeMDOPolicies
+  Checks both EOP and MDO (Safe Attachment and Safe Links) policies for user(s) specified in the CSV file or EmailAddresses parameter.
+.PARAMETER OnlyMDOPolicies
+  Checks only MDO (Safe Attachment and Safe Links) policies for user(s) specified in the CSV file or EmailAddresses parameter.
+.PARAMETER SkipConnectionCheck
+  Skips connection check for Graph and Exchange Online.
+.PARAMETER ScriptUpdateOnly
+  Just updates script version to latest one.
+
+.EXAMPLE
+	.\MDOThreatPolicyChecker.ps1
+	To check all threat policies for potentially confusing user inclusion and/or exclusion conditions and print them out for review.
+
+.EXAMPLE
+	.\MDOThreatPolicyChecker.ps1 -CsvFilePath [Path\filename.csv]
+	To provide a CSV input file with email addresses and see only EOP policies.
+
+.EXAMPLE
+	.\MDOThreatPolicyChecker.ps1 -EmailAddresses user1@contoso.com,user2@fabrikam.com
+	To provide multiple email addresses by command line and see only EOP policies.
+
+.EXAMPLE
+	.\MDOThreatPolicyChecker.ps1 -CsvFilePath [Path\filename.csv] -IncludeMDOPolicies
+	To provide a CSV input file with email addresses and see both EOP and MDO policies.
+
+.EXAMPLE
+	.\MDOThreatPolicyChecker.ps1 -EmailAddresses user1@contoso.com -OnlyMDOPolicies
+	To provide an email address and see only MDO (Safe Attachment and Safe Links) policies.
+#>
+
+[CmdletBinding(DefaultParameterSetName = 'AppliedTenant')]
+param(
+    [ValidateScript({ Test-Path $_ -PathType Leaf })]
+    [Parameter(Mandatory = $true, ParameterSetName = 'AppliedCsv')]
+    [Parameter(Mandatory = $true, ParameterSetName = 'AppliedMDOCsv')]
+    [string]$CsvFilePath,
+
+    [Parameter(ValueFromPipeline = $true, Mandatory = $true, ParameterSetName = 'AppliedEmail')]
+    [Parameter(ValueFromPipeline = $true, Mandatory = $true, ParameterSetName = 'AppliedMDOEmail')]
+    [string[]]$EmailAddresses,
```

Comment:
We should remove all plural parameter names from at least main parameters of the script itself. We should be following the guidelines described in this article: 

https://learn.microsoft.com/en-us/powershell/scripting/developer/cmdlet/strongly-encouraged-development-guidelines?view=powershell-7.4#use-singular-parameter-names

Reply:
We will make @EmailAddresses singular even though you can pass multiple ones. 
However, just to clarify, it wouldn't make sense to change IncludeMDOPolicies, ShowDetailedPolicies, or OnlyMDOPolicies to singular because the policies included or shown are always plural. You're not saying to change those too, correct?